### PR TITLE
sms/feature/integration-tests-structure

### PR DESCRIPTION
**Pull request by kalachck**

### CODE REVIEW

File: StoreManagementService/src/PBJ.StoreManagementService.Api/Extensions/IApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    public static class IApplicationBuilderExtensions
+    {
+        public static void MigrateDatabases(this IApplicationBuilder app)
+        {
+            using var serviceScope = app.ApplicationServices
+                .GetRequiredService<IServiceScopeFactory>()
+                .CreateScope();
+
+            serviceScope.ServiceProvider.GetService<DatabaseContext>()?.Database.Migrate();
+        }
```
Comment (Genresal) [inline StoreManagementService/src/PBJ.StoreManagementService.Api/Extensions/IApplicationBuilderExtensions.cs:17]: Make it async if it possible, please. I think there the `MigrateAsync` method exists.

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs [hunk 2]
```diff
@@ -109,8 +109,17 @@ public async Task<bool> DeleteAsync(int id)
...
+                await _userFollowersRepository.DeleteRangeAsync(existingUser.Followers);
+            }
+
+            if (existingUser.Followings != null)
+            {
+                await _userFollowersRepository.DeleteRangeAsync(existingUser.Followings);
+            }
 
             return true;
         }
```
Comment (Genresal) [inline StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs]: Not an issue, just a question, what is the reason to return that flag from the `delete` method?

File: StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/Managers/TestDataManager.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            _fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+        }
+
+        public async Task<Comment> CreateCommentAsync()
+        {
+            var user = _fixture.Create<User>();
+
+            _databaseContext.Users.Add(user);
+            await _databaseContext.SaveChangesAsync();
+
```
Comment (Genresal) [inline StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/Managers/TestDataManager.cs]: Why do you not reuse your method `CreateUserAsync()` there?

**Review by Genresal — APPROVED** [review]

**Review by Glipchik — APPROVED** [review]
Looks very good

**Pull request merged**